### PR TITLE
vim plugins: fix for multiple clang outputs, use clang 6.0

### PR DESCRIPTION
**Pull request by pschuprikov**
###### Motivation for this change

This PR fixes two plugins using `llvmPackages`, namely  **clang_complete** and **clighter8**, to support multiple outputs in `clang`, and changes the version of `llvmPackages` to 6.0. The main reason for the latter change is that **clighter** uses python bindings that don't work with clang 5.0.

*May be it was more appropriate to leave **clang_complete** with the default (clang 5.0) version.*

This PR is created as a more complete replacement for #45954

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/vim-plugins/vim2nix/additional-nix-code/clang_complete [hunk 1]
```diff
@@ -7,5 +7,8 @@
     # $ eval echo $(nix-instantiate --eval --expr 'with (import <nixpkgs>) {}; clang.default_cxx_stdlib_compile')
     preFixup = ''
       substituteInPlace "$out"/share/vim-plugins/clang_complete/plugin/clang_complete.vim \
```
Comment (Mic92) [inline pkgs/misc/vim-plugins/vim2nix/additional-nix-code/clang_complete]: I forgot to remove `additional-nix-code` when adding the new upgrader, we only use `override.nix` these days.

Comment (Mic92) [reply pkgs/misc/vim-plugins/vim2nix/additional-nix-code/clang_complete]: Will be removed here: https://github.com/NixOS/nixpkgs/pull/47221/files

Comment (pschuprikov) [reply pkgs/misc/vim-plugins/vim2nix/additional-nix-code/clang_complete]: Thank you, I wasn't sure whether `addiotional-nix-code` was going to be required or not. I've updated the PR not to include those changes.

**Comment by c0bw3b** [comment]
@pschuprikov could you rebase this and solve the conflict, please?

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build vimPlugins.clang_complete vimPlugins.clighter8 vimPlugins.haskell-vim

**Pull request merged**